pinot-common/src/main/java/org/apache/pinot/common/utils/ZkStarter.java

```diff
@@ -154,18 +156,23 @@ public synchronized static ZookeeperInstance startLocalZkServer(final int port,
         public void run() {
           try {
             zookeeperServerMain.initializeAndRun(args);
-          } catch (QuorumPeerConfig.ConfigException e) {
-            LOGGER.warn("Caught exception while starting ZK", e);
-          } catch (IOException e) {
+          } catch (Exception e) {
             LOGGER.warn("Caught exception while starting ZK", e);
           }
         }
       }.start();
 
       // Wait until the ZK server is started
-      ZkClient client = new ZkClient("localhost:" + port, 10000);
-      client.waitUntilConnected(10L, TimeUnit.SECONDS);
-      client.close();
+      for (int i = 0; i < 10; i++) {
```

Comment:
Don't quite follow this part. Why do we connect 10 times? It will sleep for at least 10 seconds

Reply:
the reason is that new client will fail and throw exception if zk server is unreachable unlike the previous behavior to wait to connect. So we need to handle the catch exception and retry here.

Reply:
In that case, we should break the retry if it connects successfully? Also throw exception if all 10 retries failed?

Reply:
ah, yes!
added a break after the `client.close()`